d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -867,29 +873,26 @@ public void run()
       for (ResourceType type : _resourceSubscribers.keySet())
       {
         Set<String> resources = new HashSet<>(getResourceSubscriberMap(type).keySet());
-        if (resources.isEmpty() && getWildcardResourceSubscriber(type) == null)
+        if (resources.isEmpty())
         {
           continue;
         }
-        ResourceType rewrittenType;
+
+        ResourceType rewrittenType = type;
         if (_subscribeToUriGlobCollection && type == ResourceType.D2_URI_MAP)
         {
           resources = resources.stream()
               .map(GlobCollectionUtils::globCollectionUrlForClusterResource)
               .collect(Collectors.toCollection(HashSet::new));
           rewrittenType = ResourceType.D2_URI;
         }
-        else
-        {
-          rewrittenType = type;
-        }
-        // If there is a wildcard subscriber, we should always send a wildcard request to the server.
-        if (getWildcardResourceSubscriber(type) != null)
-        {
-          resources.add("*");
-        }
         _adsStream.sendDiscoveryRequest(rewrittenType, resources);
       }
+
+      for (ResourceType type: _wildcardSubscribers.keySet())
```

Comment:
You have to use `getWildcardResourceSubscriber` here because it's how the tests are written. It overrides that function to make it return a specific map

Reply:
I noticed this today too. Will update it. thanks.

Reply:
updated